be1-go/channel/digitalCash/mod.go

```diff
@@ -0,0 +1,215 @@
+package digitalCash
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"github.com/rs/zerolog"
+	"golang.org/x/xerrors"
+	"popstellar/channel"
+	"popstellar/channel/registry"
+	"popstellar/inbox"
+	jsonrpc "popstellar/message"
+	"popstellar/message/answer"
+	"popstellar/message/messagedata"
+	"popstellar/message/query"
+	"popstellar/message/query/method"
+	"popstellar/message/query/method/message"
+	"popstellar/network/socket"
+	"popstellar/validation"
+	"strconv"
+)
+
+const (
+	msgID = "msg id"
+)
+
+// Channel is used to handle election messages.
+type Channel struct {
+	sockets   channel.Sockets
+	inbox     *inbox.Inbox
+	hub       channel.HubFunctionalities
+	channelID string
+	registry  registry.MessageRegistry
+
+	log zerolog.Logger
+}
+
+func (c *Channel) Subscribe(socket socket.Socket, msg method.Subscribe) error {
+	c.log.Info().Str(msgID, strconv.Itoa(msg.ID)).Msg("received a subscribe")
+	c.sockets.Upsert(socket)
+
+	return nil
+}
+
+func (c *Channel) Unsubscribe(socketID string, msg method.Unsubscribe) error {
+	c.log.Info().Str(msgID, strconv.Itoa(msg.ID)).Msg("received an unsubscribe")
+
+	ok := c.sockets.Delete(socketID)
+	if !ok {
+		return answer.NewError(-2, "client is not subscribed to this channel")
+	}
+
+	return nil
+}
+
+func (c *Channel) Publish(publish method.Publish, socket socket.Socket) error {
+	c.log.Info().
+		Str(msgID, strconv.Itoa(publish.ID)).
+		Msg("received a publish")
+
+	err := c.verifyMessage(publish.Params.Message)
+	if err != nil {
+		return xerrors.Errorf("failed to verify publish message on an "+
+			"election channel: %w", err)
+	}
+
+	err = c.handleMessage(publish.Params.Message, socket)
+	if err != nil {
+		return xerrors.Errorf("failed to handle a publish message: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Channel) Catchup(catchup method.Catchup) []message.Message {
+	c.log.Info().Str(msgID, strconv.Itoa(catchup.ID)).Msg("received a catchup")
+
+	return c.inbox.GetSortedMessages()
+}
+
+func (c *Channel) Broadcast(broadcast method.Broadcast, socket socket.Socket) error {
+	c.log.Info().Msg("received a broadcast")
+
+	err := c.verifyMessage(broadcast.Params.Message)
+	if err != nil {
+		return xerrors.Errorf("failed to verify broadcast message: %w", err)
+	}
+
+	err = c.handleMessage(broadcast.Params.Message, socket)
+	if err != nil {
+		return xerrors.Errorf("failed to handle broadcast message: %v", err)
+	}
+
+	return nil
+}
+
+// NewChannel returns a new initialized digitalCash channel
+func NewChannel(channelID string, log zerolog.Logger, hub channel.HubFunctionalities) channel.Channel {
```

Comment:
You should probably make your channel follow the organization of the other channels